tests/functional/aws-node-sdk/test/object/mpu.js

```diff
@@ -39,45 +75,66 @@ describe('aws-node-sdk test suite of listMultipartUploads', () => {
             return s3.createBucketAsync({ Bucket: bucket })
             .then(() => bucketUtil.getOwner())
             .then(res => {
-                // In this case, the owner of the bucket will also be the MPU
-                // upload owner. We need this value for testing comparison.
-                displayName = res.DisplayName;
+                // The owner of the bucket will also be the MPU upload owner.
+                data.displayName = res.DisplayName;
+                data.userId = res.ID;
             })
             .then(() => s3.createMultipartUploadAsync({
                 Bucket: bucket,
                 Key: objectKey,
             }))
             .then(res => {
-                uploadId = res.UploadId;
-            })
-            .catch(err => throwErr('Error in beforeEach', err));
+                data.uploadId = res.UploadId;
+            });
         });
 
         afterEach(() =>
             s3.abortMultipartUploadAsync({
                 Bucket: bucket,
                 Key: objectKey,
-                UploadId: uploadId,
+                UploadId: data.uploadId,
             })
             .then(() => bucketUtil.empty(bucket))
             .then(() => bucketUtil.deleteOne(bucket))
-            .catch(err => throwErr('Error in afterEach', err))
         );
 
-        it('should list ongoing multipart uploads', done => {
+        it('should list ongoing multipart uploads', () =>
             s3.listMultipartUploadsAsync({ Bucket: bucket })
-            .then(res => checkValues(res, uploadId, displayName, done))
-            .catch(done);
-        });
+            .then(res => checkValues(res, data))
+        );
 
-        it('should list ongoing multipart uploads with params', done => {
-            s3.listMultipartUploadsAsync({
+        it('should list ongoing multipart uploads with params', () => {
+            data.prefixVal = 'to';
+            data.delimiter = 'test-delimiter';
+            data.maxUploads = 1;
+
+            return s3.listMultipartUploadsAsync({
                 Bucket: bucket,
                 Prefix: 'to',
+                Delimiter: 'test-delimiter',
                 MaxUploads: 1,
             })
-            .then(res => checkValues(res, uploadId, displayName, done))
-            .catch(done);
+            .then(res => checkValues(res, data));
         });
-    });
-});
+
+        it('should list 0 multipart uploads when MaxUploads is 0', () =>
+            s3.listMultipartUploadsAsync({
+                Bucket: bucket,
+                MaxUploads: 0,
+            })
+            .then(res =>
+                // When `MaxUploads` is set to 0 there is no `Initiated` value.
+                // Also, `IsTruncated` is set to `false` despite the fact that
+                // there is, in fact, one multipart upload in the bucket.
+                assert.deepStrictEqual(res, {
```

Comment:
I had decided the response object was different enough when `MaxUploads` is 0 that it warranted its own object. It would definitely work with the compare function though, and I think that would separate concerns better. Updated! 😃